src/DepositHandler.sol

```diff
@@ -133,28 +122,30 @@ contract DepositHandler is Pausable, AccessControl, IDepositHandlerErrors {
     }
 
     /**
-     * @notice Allow users to withdraw their deposits if some emergency situation appear.
+     * @notice Automatically withdraw users deposits back to them if some emergency situation appear.
      * Example of emergency situation:
      *  1. Manager set up incorrect(too long duration).
      *  2. Smth happens on Encode side.
-     * Which situation isn't a point to call this function:
-     *  1. If protocol was hacked this function can't be called, because in this scenario
-     * protocol will be on pause and all critical functions will be blocked for usage until
-     * problem will be resolved.
+     *  3. Stange activity mentioned in the contract.
      * @dev Allow to withdraw user funds if an emergency situation appear.
      * Function restrictions:
+     *  - Can only be called when contract is not on Pause.
      *  - Can only be called when `allowEmergencyWithdraw` is true.
+     *  - Can only be called by `MANAGER`.
      * Function omits the next checks:
      *  - No check whether the user has a `Withdraw` status.
-     *  - No check for the bootcamp finality.
-     * @param _amount - amount to withdraw.
-     * @param _depositor - address which will receive a deposit back.
      */
-    function emergencyWithdraw(uint256 _amount, address _depositor) external {
+    function emergencyWithdraw() external onlyRole(MANAGER) {
```

Comment:
We will need to test that and make sure the gas cost isn't too high, and optimize it for 50 people and make it clear it may not be optimized for more

Reply:
Yes 100% we need this for all the functions which is using a for loop